corehq/apps/enterprise/iterators.py

```diff
@@ -1,7 +1,242 @@
+      the previous progress arguments
+    - start_date: a date to start the date range. Can be None
+    - end_date: the inclusive date to finish the date range. Can be None
+    last_domain, last_time, and last_id are intended to represent the last result from a previous query
+    that should now be resumed.
+    - last_domain: the domain to resume the query on
+    - last_time: the timestamp from the last result of the previous query
+    - last_id: the id from the last result of the previous query
+    '''
+    def __init__(self, account, form_converter, start_date, end_date, last_domain, last_time, last_id):
+        MAX_DATE_RANGE_DAYS = 100
+        (self.start_date, self.end_date) = resolve_start_and_end_date(start_date, end_date, MAX_DATE_RANGE_DAYS)
+        self.account = account
+        self.form_converter = form_converter
+        self.last_domain = last_domain
+        self.last_time = last_time
+        self.last_id = last_id
+
+    def execute(self, limit=None):
+        domains = self.account.get_domains()
+
+        it = run_query_over_domains(
+            MobileFormSubmissionsQueryFactory(),
+            domains,
+            limit=limit,
+            last_domain=self.last_domain,
+            start_date=self.start_date,
+            end_date=self.end_date,
+            last_time=self.last_time,
+            last_id=self.last_id,
+        )
+
+        return (self.form_converter.convert(form) for form in it)
+
+    def get_query_params(self, fetched_object):
+        return self.form_converter.get_query_params(fetched_object)
+
+
+def resolve_start_and_end_date(start_date, end_date, maximum_date_range):
+    '''
+    Provide start and end date values if not supplied.
+    '''
+    if not end_date:
+        end_date = datetime.now(timezone.utc)
+
+    if not start_date:
+        start_date = end_date - timedelta(days=30)
+
+    if end_date - start_date > timedelta(days=maximum_date_range):
+        raise TooMuchRequestedDataError(
+            _('Date ranges with more than {} days are not supported').format(maximum_date_range)
+        )
+
+    return start_date, end_date
+
+
+class EnterpriseFormReportConverter:
+    def __init__(self):
+        self.app_lookup = AppIdToNameResolver()
+
+    def convert(self, form):
+        domain = form['domain']
+        submitted_date = datetime.strptime(form['received_on'][:19], '%Y-%m-%dT%H:%M:%S')
+        inserted_at = datetime.strptime(form['inserted_at'][:19], '%Y-%m-%dT%H:%M:%S')
+
+        return {
+            'form_id': form['form']['meta']['instanceID'],
+            'form_name': form['form']['@name'] or _('Unnamed'),
+            'submitted': submitted_date,
+            'inserted_at': inserted_at,
+            'app_name': self.app_lookup.resolve_app_id_to_name(domain, form['app_id']) or _('App not found'),
+            'username': form['form']['meta']['username'],
+            'domain': domain
+        }
+
+    @classmethod
+    def get_query_paraams(cls, fetched_object):
```

Comment:
`get_query_params` instead of `paraaaaaaaams`

Reply:
tests were passing with this incorrect name. good to verify if tests are covering this?

Reply:
Verified that the reason this was not flagged on tests was because `get_query_params` is only called for queries that extend beyond page boundaries. The test in _test_apis_ is intended to just be a happy path test, and so doesn't try to verify more than the base case -- here, it constructs just 2 forms. A test does exist to ensure that the paginator handles page boundaries correctly, but its generic (in _keyset_paginator_tests_). I feel like this is mostly a typo error on my part, and we probably don't need multiple integration tests to verify each functionality of the APIs. If I were to add a test for form submission page boundaries, that would mean I'd want to do the same for every new API added.